src/text/raw_text_reader.rs

```diff
@@ -249,25 +266,52 @@ impl<T: TextIonDataSource> RawTextReader<T> {
         }
     }
 
+    fn parse_next<P, O>(&mut self, parser: P) -> IonResult<Option<O>>
+    where
+        P: Fn(&str) -> IonParseResult<O>,
+    {
+        match self.parse_next_nom(parser) {
```

Comment:
🗺️ `parse_next` is now an adapter around `parse_next_nom` below. It takes the parse result and maps it to a user-facing `IonResult`.